File: scala/digital-wallet/src/main/scala/core/domain/model/CreateJournalEntry.scala
package core.domain.model

import core.domain.enums.BalanceType.BalanceType
import core.domain.enums.SubwalletType.SubwalletType

case class CreateJournalEntry(
  walletId: Option[String],
  subwalletType: SubwalletType,
  balanceType: BalanceType,
  amount: BigDecimal,
)

File: kotlin/digital-wallet/src/main/kotlin/digitalwallet/core/domain/models/CreateJournalEntry.kt
package digitalwallet.core.domain.models

import digitalwallet.core.domain.enums.BalanceType
import digitalwallet.core.domain.enums.SubwalletType
import java.math.BigDecimal

data class CreateJournalEntry(
    val walletId: String?,
    val subwalletType: SubwalletType,
    val balanceType: BalanceType,
    val amount: BigDecimal,
)